typed/rosette/base-forms.rkt

```diff
@@ -123,9 +123,10 @@
         lam)]]
   ;; function with type declaration beforehand
   [(_ (f:id/type-decl . (~and args (:id ... (~seq :keyword [:id :expr]) ...))) body ...+) ≫
+   #:with body* (syntax/loc this-syntax (begin body ...))
+   #:with lam (syntax/loc this-syntax lam)
```

Comment:
Don't rush.

I gave you push access so go ahead and merge when you're done and the tests pass. Thanks!